### ISSUE TITLE
does sdxl lora training train both text encoders or only 1?

### ISSUE THREAD

**Issue by DarkViewAI** [bmaltais/kohya_ss#2639]
results with dreambooth and text encoders are pretty unstable and unreliable. but when i train via lora and with the text encoder results are pretty good. so i'm wondering if lora training only trains one text encoder or both.

Thanks!

**Comment by b-fission** [comment]
Do you know if your dreambooth config is actually training the text encoders? 

Current versions of kohya_ss seem to have it off by default so I'd specify `--train_text_encoder` whenever I use dreambooth. See if `text_encoder1` and `text_encoder2` are false in your log.

```
INFO     caching latents...
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 116/116 [00:40<00:00,  2.86it/s]
train unet: True, text_encoder1: False, text_encoder2: False
number of models: 1
number of trainable parameters: 2567463684
prepare optimizer, data loader etc.
```

**Comment by bmaltais** [comment]
The way the GUI work is if you don't specify a text encoder learning value it will not get train. Now, regarding the training of both text encoders for LoRA... I don't know... what I know is that it only allow to specify a single text encoder learning rate value... Only SDXL finetuning allow to specify a learning rate for both...

I suggest you ask the question directly on kohya's sd-scripts repo to get the complete answer.

**Comment by DarkViewAI** [comment]
@b-fission yep im aware about dreambooth, it seems like lora training also does both text encoders and not 1
